File: Wprowadzenie do funkcji/IV. Wprowadzenie do funkcji/zadanie2-rozwiazanie.cpp
#include <iostream>
using namespace std;

bool zawiera_wyraz(string tekst, string wyraz)
{
    for(int i = 0; i < tekst.length() - wyraz.length() + 1; i++)
    {
        bool ok = true;
        for(int j = 0; j < wyraz.length(); j++)
        {
            if(tekst[i + j] != wyraz[j])
            {
                ok = false;
                break;
            }
        }

        if(ok)
        {
            return true;
        }
    }

    return false;
}

string odwroc_wyraz(string wyraz) {
    string odwrocony = "";
    for(int i = 0; i < wyraz.length(); i++) {
        odwrocony = wyraz[i] + odwrocony;
    }

    return odwrocony;
}

int main()
{
    string tekst, wyraz;
    cin >> tekst >> wyraz;

    string odwrocony;
    odwrocony = odwroc_wyraz(wyraz);

    if(zawiera_wyraz(tekst, wyraz)) {
        cout << "Obraza I stopnia" << endl;
    } else if(zawiera_wyraz(tekst, odwrocony)) {
        cout << "Obraza II stopnia" << endl;
    } else {
        cout << "Brak obrazy" << endl;
    }
    
    return 0;
}

File: Wprowadzenie do funkcji/V. Wprowadzenie do funkcji/zadanie2-rozwiazanie.py
def zawiera_wyraz(tekst, wyraz):
    for i in range(0, len(tekst) - len(wyraz) + 1):
        ok = True
        for j in range(0, len(wyraz)):
            if tekst[i+j] != wyraz[j]:
                ok = False
                break
        if ok:
            return True
    return False

def odwroc_wyraz(wyraz):
    odwrocony = ""
    for i in range(0, len(wyraz)):
        odwrocony = wyraz[i] + odwrocony
    return odwrocony


tekst = input()
wyraz = input()

odwrocony = odwroc_wyraz(wyraz)
if zawiera_wyraz(tekst, wyraz):
    print("Obraza I stopnia")
elif zawiera_wyraz(tekst, odwrocony):
    print("Obraza II stopnia")
else:
    print("Brak obrazy")